resolwe/flow/views/mixins.py

```diff
@@ -102,7 +103,55 @@ def update(self, request, *args, **kwargs):
                         'Invalid descriptor_schema slug "{}" - object does not exist.'.format(ds_query)]},
                     status=status.HTTP_400_BAD_REQUEST)
 
-        return super(ResolweUpdateModelMixin, self).update(request, *args, **kwargs)
+        # return super(ResolweUpdateModelMixin, self).update(request, *args, **kwargs)
+        return self._update(request, *args, **kwargs)
+
+    # NOTE: This is a copy of rest_framework.mixins.UpdateModelMixin.update().
+    #       The self.get_object() was replaced with the
+    #       self.get_object_with_lock() to lock the object while updating.
+    #       Use the original method when suport for locking is added:
+    #       https://github.com/encode/django-rest-framework/issues/4675
+    def _update(self, request, *args, **kwargs):
+        """Update a resource."""
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object_with_lock()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+
+        if getattr(instance, '_prefetched_objects_cache', None):
+            # If 'prefetch_related' has been applied to a queryset, we need to
+            # forcibly invalidate the prefetch cache on the instance.
+            instance._prefetched_objects_cache = {}  # pylint: disable=protected-access
+
+        return Response(serializer.data)
+
+    # NOTE: This is a copy of rest_framework.generics.GenericAPIView.get_object().
+    #       The select_for_update() was added to the 'queryset'
+    #       to lock the object while updating.
+    #       Use the original method when suport for locking is added:
+    #       https://github.com/encode/django-rest-framework/issues/4675
+    def get_object_with_lock(self):
+        """Return the object the view is displaying."""
+        queryset = self.filter_queryset(self.get_queryset())
+
+        # Perform the lookup filtering.
+        lookup_url_kwarg = self.lookup_url_kwarg or self.lookup_field
+
+        assert lookup_url_kwarg in self.kwargs, (
+            'Expected view %s to be called with a URL keyword argument '
+            'named "%s". Fix your URL conf, or set the `.lookup_field` '
+            'attribute on the view correctly.' %
+            (self.__class__.__name__, lookup_url_kwarg)
+        )
+
+        filter_kwargs = {self.lookup_field: self.kwargs[lookup_url_kwarg]}
+        obj = get_object_or_404(queryset.select_for_update(), **filter_kwargs)
```

Comment:
I think it would be nice to explicitly mark the changed line, even though this is mentioned in the comment above.